client-impl/mapreduce-client/src/main/java/uk/gov/gchq/palisade/mapreduce/PalisadeInputSplit.java

```diff
@@ -117,7 +117,8 @@ public long getLength() throws IOException, InterruptedException {
     public void write(final DataOutput dataOutput) throws IOException {
         Objects.requireNonNull(dataOutput, "dataOutput");
```

Comment:
Can you add a more detailed error message, such as "The data output cannot be set to null."